omniduct/filesystems/s3.py

```diff
@@ -189,11 +189,27 @@ def _dir(self, path):
     #         if pattern is None or pattern.match(k.key[len(path_prefix):]):
     #             yield k.key
 
-    def _mkdir(self, path, recursive):
-        raise NotImplementedError("Support for S3 write operations has yet to be implemented.")
+    def _mkdir(self, path, recursive, exist_ok):
+        if not path.endswith('/'):
+            path += '/'
+        if not self._exists(path):
+            self._client.put_object(Bucket=self.bucket, Key=self._path(path))
+
+    def _remove_file(self, path):
+        """Removes a single key or file from s3"""
+        s3 = self._resource
+        s3.Object(self.bucket, self._path(path)).delete()
+        return True
 
     def _remove(self, path, recursive):
-        raise NotImplementedError("Support for S3 deletion operations has yet to be implemented.")
+        s3 = self._resource
+        bucket = s3.Bucket(self.bucket)
+        if not recursive:
+            return self._remove_file(path)
+        else:
+            for obj in bucket.objects.filter(Prefix=self._path(path)):
```

Comment:
This may have unintended side-effects if the path matches another path partially. Perhaps first check if the path exists?

Reply:
made the change to check if the path exists then it'll find if it's a dir or file then will delete accordingly.